.github/workflows/failureNotifier.yml

```diff
@@ -27,19 +27,52 @@ jobs:
             });
             return jobsData.data;
 
+      - name: Fetch Previous Workflow Run
+        id: previous-workflow-run
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const runId = ${{ github.event.workflow_run.id }};
+            const allRuns = await github.rest.actions.listWorkflowRuns({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: 'preDeploy.yml',
+            });
+            const run = allRuns.data.workflow_runs.find(run => run.id === runId);
+            const run_number = run.run_number;
+            const previousRun = allRuns.data.workflow_runs.find(run => run.run_number === run_number - 1);
+            if (previousRun.actor.login === 'OSBotify') {
+              return allRuns.data.workflow_runs.find(run => run.run_number === run_number - 2);
+            }
```

Comment:
I don't think this will work if multiple runs are from `OSBotify`. Can we just filter out any runs with `actor.login === 'OSBotify'` first before getting the last one? Do we also need to filter out cancelled runs?